engine/src/main/java/org/pentaho/di/job/JobExecutionConfiguration.java

```diff
@@ -0,0 +1,720 @@
+    for ( int i = 0; i < nrArguments; i++ ) {
+      Node argNode = XMLHandler.getSubNodeByNr( argsNode, "argument", i );
+      String name = XMLHandler.getTagValue( argNode, "name" );
+      String value = XMLHandler.getTagValue( argNode, "value" );
+      if ( !Utils.isEmpty( name ) && !Utils.isEmpty( value ) ) {
+        arguments.put( name, value );
+      }
+    }
+
+    // Read the parameters...
+    //
+    Node parmsNode = XMLHandler.getSubNode( trecNode, "parameters" );
+    int nrParams = XMLHandler.countNodes( parmsNode, "parameter" );
+    for ( int i = 0; i < nrParams; i++ ) {
+      Node parmNode = XMLHandler.getSubNodeByNr( parmsNode, "parameter", i );
+      String name = XMLHandler.getTagValue( parmNode, "name" );
+      String value = XMLHandler.getTagValue( parmNode, "value" );
+      if ( !Utils.isEmpty( name ) ) {
+        params.put( name, value );
+      }
+    }
+
+    replayDate = XMLHandler.stringToDate( XMLHandler.getTagValue( trecNode, "replay_date" ) );
+    safeModeEnabled = "Y".equalsIgnoreCase( XMLHandler.getTagValue( trecNode, "safe_mode" ) );
+    logLevel = LogLevel.getLogLevelForCode( XMLHandler.getTagValue( trecNode, "log_level" ) );
+    clearingLog = "Y".equalsIgnoreCase( XMLHandler.getTagValue( trecNode, "clear_log" ) );
+
+    startCopyName = XMLHandler.getTagValue( trecNode, "start_copy_name" );
+    startCopyNr = Const.toInt( XMLHandler.getTagValue( trecNode, "start_copy_nr" ), 0 );
+
+    gatheringMetrics = "Y".equalsIgnoreCase( XMLHandler.getTagValue( trecNode, "gather_metrics" ) );
+
+    String sPassedBatchId = XMLHandler.getTagValue( trecNode, "passedBatchId" );
+    if ( !StringUtils.isEmpty( sPassedBatchId ) ) {
+      passedBatchId = Long.parseLong( sPassedBatchId );
+    }
+
+    Node resultNode = XMLHandler.getSubNode( trecNode, Result.XML_TAG );
+    if ( resultNode != null ) {
+      try {
+        previousResult = new Result( resultNode );
+      } catch ( KettleException e ) {
+        throw new KettleException( "Unable to hydrate previous result", e );
+      }
+    }
+
+    // Try to get a handle to the repository from here...
+    //
+    Node repNode = XMLHandler.getSubNode( trecNode, "repository" );
+    if ( repNode != null ) {
+      String repositoryName = XMLHandler.getTagValue( repNode, "name" );
+      String username = XMLHandler.getTagValue( repNode, "login" );
+      String password = Encr.decryptPassword( XMLHandler.getTagValue( repNode, "password" ) );
+      connectRepository( repositoryName, username, password );
+    }
+
+  }
+
+  public Repository connectRepository( String repositoryName, String username, String password ) throws KettleException {
+    // Verify that the repository exists on the slave server...
+    //
+    RepositoriesMeta repositoriesMeta = new RepositoriesMeta();
+    repositoriesMeta.getLog().setLogLevel( log.getLogLevel() );
+    try {
+      repositoriesMeta.readData();
+    } catch ( Exception e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)